spec/models/post_spec.rb

```diff
@@ -0,0 +1,63 @@
+# frozen_string_literal: true
+
+# == Schema Information
+#
+# Table name: posts
+#
+#  id         :bigint           not null, primary key
+#  body       :text             not null
+#  status     :integer          default("public"), not null
+#  created_at :datetime         not null
+#  updated_at :datetime         not null
+#  user_id    :bigint
+#
+# Indexes
+#
+#  index_posts_on_user_id  (user_id)
+#
+require 'rails_helper'
+
+RSpec.describe Post, type: :model do
+  context 'when everything is valid' do
+    let(:post) { FactoryBot.create :post }
+    let(:private) { FactoryBot.build :post, :private }
+
+    it 'is expected to be private post' do expect(private).to be_valid end
```

Comment:
`it 'is expected to be private post'` - I wonder how do you translate it to Russian :D
1. it would be correct to say `it 'expects private post to be valid'`
2. if you provide text to `it` block, this text is used for generating spec message
but `expect(private).to be_valid` already generates `is expected to be valid`
So without providing text to `it` block, you already have pretty nice spec message tree (though still a bit funny, when everything is valid - it is valid...)
```
Post
  when everything is valid
    is expected to be valid
```
3. I don't like testing regular post and private post here. This context is for testing regular post validations, associations etc. If you want to emphasize that private post has same rules/checks/etc as regular post. you should write separate specs for private post. You can use _shared examples_ for that and just run it for regular post & for private post. Read about rspec shared examples and shared context.